lab_3_ann_retriever/main.py

```diff
@@ -157,6 +190,22 @@ def build(self) -> bool:
         Returns:
             bool: True if built successfully, False in other case
         """
+        if not self._corpus or not isinstance(self._corpus,list):
+            return False
+        self._vocabulary = sorted(list({word for text in self._corpus for word in text}))
+        for word in self._vocabulary:
+            word_instances = 0
+            for text in self._corpus:
+                if word in text:
+                    word_instances += 1
+            self._idf_values[word] = math.log((len(self._corpus) - word_instances + 0.5) / (word_instances + 0.5))
```

Comment:
use `calculate_idf` from lab_2

Reply:
used

Reply:
importing idf from lab2 broke my start.py. Is it supposed to do that or did something go wrong on my part? removed one vectorizer.build() line from start and it seemed to fix it, but i'm not sure if that's what i'm supposed to do with that